account-kit/smart-contracts/src/ma-v2/client/client.test.ts

```diff
@@ -329,7 +336,235 @@ describe("MA v2 Tests", async () => {
     );
   });
 
-  it("installs a session key via deferred action signed by the owner and has it sign a UO", async () => {
+  it("installs a session key via deferred action using PermissionBuilder signed by the owner and has it sign a UO", async () => {
+    let provider = (
+      await givenConnectedProvider({
+        signer,
+      })
+    )
+      .extend(installValidationActions)
+      .extend(deferralActions);
+
+    await setBalance(client, {
+      address: provider.getAddress(),
+      value: parseEther("2"),
+    });
+
+    const { entityId, nonce } = await provider.getEntityIdAndNonce({
+      entityId: 0,
+      nonceKey: 0n,
+      isGlobalValidation: false,
+    });
+
+    // Must be built with the client that's going to sign the deferred action
+    // OR a client with the same set signer entity as the signing client (entityId + isGlobal)
+    const { typedData, fullPreSignatureDeferredActionDigest } =
+      await new PermissionBuilder(provider)
+        .configure({
+          key: {
+            publicKey: await sessionKey.getAddress(),
+            type: "secp256k1",
+          },
+          entityId: entityId,
+          nonce: nonce,
+        })
+        .addPermission({
+          permission: {
+            type: PermissionType.GAS_LIMIT,
+            data: {
+              limit: toHex(parseEther("1")),
+            },
+          },
+        })
+        .addPermission({
+          permission: {
+            type: PermissionType.ACCOUNT_FUNCTIONS,
+            data: {
+              functions: ["0xb61d27f6"], // execute selector
+            },
+          },
+        })
+        .compile_deferred({
+          deadline: 0,
+          uoValidationEntityId: entityId,
+          uoIsGlobalValidation: false,
+        });
+
+    // Sign the typed data using the owner (fallback) validation, this must be done via the account to skip 6492
+    const deferredValidationSig = await provider.account.signTypedData(
+      typedData
+    );
+
+    // Build the full hex to prepend to the UO signature
+    const deferredActionDigest = provider.buildDeferredActionDigest({
```

Comment:
so this is an action on the client itself? does it need to be a client action? can it be a static method on the account type or even just an account instance method?